src/Service/NoHitService.php

```diff
@@ -17,15 +18,25 @@
  */
 class NoHitService
 {
+    private $noHitsProcessingEnabled;
     protected $dispatcher;
 
     /**
      * NoHitService constructor.
      *
+     * @param ParameterBagInterface $params
+     *   Access to environment variables
      * @param EventDispatcherInterface $dispatcher
+     *   Event dispatcher
      */
-    public function __construct(EventDispatcherInterface $dispatcher)
+    public function __construct(ParameterBagInterface $params, EventDispatcherInterface $dispatcher)
```

Comment:
Should this have been changed to an $bindNoHitsProcessingEnabled variable to remove the ParameterBagInterface could default to true in the parameter.

Reply:
As above. I considered a switch to be out of scope for this PR